python/cuml/experimental/common/base.pyx

```diff
@@ -0,0 +1,555 @@
+                continue
+            if type(state[key]) is str:
+                string += "{}='{}', ".format(key, state[key])
+            else:
+                if hasattr(state[key], "__str__"):
+                    string += "{}={}, ".format(key, state[key])
+        string = string.rstrip(', ')
+        output = string + ')'
+
+        if hasattr(self, 'sk_model_'):
+            output += ' <sk_model_ attribute used>'
+        return output
+
+    def dispatch_func(self, func_name, *args, **kwargs):
+        """
+        This function will dispatch calls to training and inference according
+        to the global configuration. It should work for all estimators
+        sufficiently close the scikit-learn implementation as it uses
+        it for training and inferences on host.
+
+        Parameters
+        ----------
+        func_name : string
+            name of the function to be dispatched
+        args : arguments
+            arguments to be passed to the function for the call
+        kwargs : keyword arguments
+            keyword arguments to be passed to the function for the call
+        """
+        # look for current device_type
+        device_type = cuml.global_settings.device_type
+        if device_type == DeviceType.device:
+            # call the original cuml method
+            cuml_func_name = '_' + func_name
+            if hasattr(self, cuml_func_name):
+                cuml_func = getattr(self, cuml_func_name)
+                return cuml_func(*args, **kwargs)
+            else:
+                raise ValueError('Function "{}" could not be found in'
+                                 ' the cuML estimator'.format(cuml_func_name))
+
+        elif device_type == DeviceType.host:
+            # check if the sklean model already set as attribute of the cuml
+            # estimator its presence should signify that CPU execution was
+            # used previously
+            if not hasattr(self, 'sk_model_'):
+                # import model in sklearn
+                if hasattr(self, 'sk_import_path_'):
+                    # if import path differs from the one of sklearn
+                    # look for sk_import_path_
+                    model_path = self.sk_import_path_
+                else:
+                    # import from similar path to the current estimator
+                    # class
+                    model_path = 'sklearn' + self.__class__.__module__[4:]
+                model_name = self.__class__.__name__
+                sk_model = getattr(import_module(model_path), model_name)
+                # initialize model
+                self.sk_model_ = sk_model()
+                # transfer params set during cuml estimator initialization
+                for param in self.get_param_names():
+                    self.sk_model_.__dict__[param] = self.__dict__[param]
+
+                # transfer attributes trained with cuml
+                for attr in self.get_attributes_names():
```

Comment:
Yep, this makes sense. I'm assuming `attribute_names` in this context means the trained parameters that define the model? That should be pretty easy to match in most cases.